mesonbuild/backend/backends.py

```diff
@@ -1540,6 +1540,10 @@ def guess_install_tag(self, fname: str, outdir: T.Optional[str] = None) -> T.Opt
             return 'devel'
         elif localedir in dest_path.parents:
             return 'i18n'
+        elif 'installed-tests' in dest_path.parts:
```

Comment:
Shouldn't we also check that it's a `datadir`?

Reply:
No, they seems to be all over the places, on my Ubuntu:
```
$ find /usr -name installed-tests
/usr/lib/x86_64-linux-gnu/installed-tests
/usr/share/installed-tests
/usr/libexec/installed-tests
```

Maybe could restrict to `[datadir | libdir | libexecdir]/installed-tests`, but that seemed easier and safe enough to assume that anything with `installed-tests` in their path should be tagged as a test by default.

Reply:
Note that this is just Meson doing guesses based on heuristics in the case projects did not specify an explicit `install_tag` kwarg. So the project can always override that.

Reply:
`${datadir}/installed-tests` is the directory for test metadata (the .ini-style files listing the available tests) in GNOME's installed-tests specification. That's the only path that is "API".

`${libexecdir}/installed-tests` is the conventional directory for the actual tests (test executables, helper executables, test data, etc.). The specification specifically says that this is only conventional and not a functional requirement (but it's as good a convention as any other, and seems entirely reasonable for a heuristic like this).

`${libdir}/installed-tests` is *mostly* a Debianism: Debian and Ubuntu sometimes install the actual tests into that location instead of in the `${libexecdir}`, so that you can have both x86_64 and i386 tests installed at the same time on the same system. However, it's not entirely Debian-specific: projects that care a lot about Red-Hat-style multilib (`lib64` and `lib`) or Arch-style multilib (`lib` and `lib32`) might want to do the same.